lib/SILOptimizer/Utils/ConstantFolding.cpp

```diff
@@ -278,6 +278,44 @@ constantFoldBinaryWithOverflow(BuiltinInst *BI, BuiltinValueKind ID,
            ResultsInError);
 }
 
+static SILValue countZeros(BuiltinInst *BI, llvm::Intrinsic::ID ID) {
+  assert(BI->getArguments().size() == 2 && "Ctlz should have 2 args.");
+  OperandValueArrayRef Args = BI->getArguments();
+
+  // Fold for integer constant arguments.
+  auto *LHS = dyn_cast<IntegerLiteralInst>(Args[0]);
+  if (!LHS) {
+    return nullptr;
+  }
+  APInt LHSI = LHS->getValue();
+  unsigned LZ = 0;
+  // Check corner-case of source == zero
+  if (LHSI == 0) {
+    auto *RHS = dyn_cast<IntegerLiteralInst>(Args[1]);
+    if (!RHS || RHS->getValue() != 0) {
+      // Undefined
+      return nullptr;
+    }
+    LZ = LHSI.getBitWidth();
+  } else {
+    switch (ID) {
```

Comment:
My suggestion: Use an inline constructor:

unsigned LZ = [&] {
   ...
}();

Also. Do you really think we are going to support more types of intrinsics here? Why not just put in an assert or something like that.